### PR TITLE
Active storage

### PR DESCRIPTION
**Pull request by tktezy**

### CODE REVIEW

File: app/views/comments/_comment.html.erb [hunk 1]
```diff
@@ -11,6 +11,17 @@
 				</div>
 			<% end %>
 		</div>
-		<span class="d-block"><%= comment.body %></span>
+		<span class="d-block">
+			<%= comment.body %>
+			<% (0...comment.images.count).each do |image| %>
```
Comment (AlexeyAndrianov) [inline app/views/comments/_comment.html.erb]: I suggest you to use Boleyn methods like if comment.images.any?

File: app/views/comments/_comment.html.erb [hunk 2]
```diff
@@ -11,6 +11,17 @@
...
+				<div style="width: 30%; height: 30%">
+					<%= image_tag(comment.images[image], style:'width:100%; height:auto') %>
+				</div>
+			<% end %>
+			<br>
+			<% (0...comment.files.count).each do |file| %>
```
Comment (AlexeyAndrianov) [inline app/views/comments/_comment.html.erb]: Views are becoming too complex. Maybe it’s time to use decorators?

File: app/models/ability.rb [hunk 3]
```diff
@@ -16,14 +16,13 @@ def initialize(user)
       can :destroy, Comment do |comment|
         comment.user == user
       end
+      cannot :read, User
     #client only can read
     elsif user.has_role? :client
-      can :read, :all
+      cannot :read, User
     elsif user.has_role? :manager
-      can :read, :all
```
Comment (AlexeyAndrianov) [inline app/models/ability.rb]: Have you tried testing this by logging with non-admin user?

File: app/models/project.rb [hunk 4]
```diff
@@ -2,5 +2,6 @@ class Project < ApplicationRecord
 	belongs_to :user
 	has_many   :tasks, dependent: :destroy
 	validates  :project_title, presence: true
```
Comment (AlexeyAndrianov) [inline app/models/project.rb]: Maybe this should be an image (avatar)

File: app/views/projects/show.html.erb [hunk 5]
```diff
@@ -7,7 +7,7 @@
 		| <%= link_to 'Удалить', project_path(@project), method: :delete, data: { confirm: 'Вы уверены?'} %>
 	</p>
 <% end %>
-
+<%= link_to 'Скачать файл', @project.file %>
```
Comment (AlexeyAndrianov) [inline app/views/projects/show.html.erb]: How do we store files? Locally? I suggest using remote services

File: app/views/comments/_comment.html.erb (hunk 2 above)
Comment (AlexeyAndrianov) [reply app/views/comments/_comment.html.erb]: Or something else)

File: app/views/projects/show.html.erb (hunk 5 above)
Comment (AlexeyAndrianov) [inline app/views/projects/show.html.erb]: what if we dont have any files? rails app will crash. handle this

File: app/models/ability.rb [hunk 6]
```diff
@@ -16,14 +16,13 @@ def initialize(user)
       can :destroy, Comment do |comment|
         comment.user == user
       end
+      cannot :read, User
     #client only can read
     elsif user.has_role? :client
-      can :read, :all
```
Comment (AlexeyAndrianov) [inline app/models/ability.rb]: client now can read from all the projects. is this ok?

File: app/models/ability.rb (hunk 3 above)
Comment (tktezy) [reply app/models/ability.rb]: in def_roles branch i changed this to cannot :read, User

Comment (tktezy) [reply app/models/ability.rb]: and i tested it

File: app/views/projects/show.html.erb (hunk 5 above)
Comment (tktezy) [reply app/views/projects/show.html.erb]: ill fix it

File: app/models/ability.rb (hunk 6 above)
Comment (tktezy) [reply app/models/ability.rb]: see def_roles branch

File: app/views/projects/show.html.erb (hunk 5 above)
Comment (tktezy) [reply app/views/projects/show.html.erb]: temporarily locally

File: app/views/comments/_comment.html.erb (hunk 1 above)
Comment (tktezy) [reply app/views/comments/_comment.html.erb]: didnt get it

Comment (AlexeyAndrianov) [reply app/views/comments/_comment.html.erb]: @tktezy if comment.images,present?
will be much more readable than iteration starting with 0

**Pull request merged**